core/src/lib/jcr.cc

```diff
@@ -216,14 +216,19 @@ JobControlRecord* new_jcr(JCR_free_HANDLER* daemon_free_jcr)
   jcr = new (jcr) JobControlRecord();
 
   jcr->daemon_free_jcr = daemon_free_jcr;
+  return jcr;
+}
+
+void register_jcr(JobControlRecord* jcr)
+{
+  Dmsg0(debuglevel, "Enter register_jcr\n");
```

Comment:
The reason for this two-step initialization is because consumers of the jcr chain sometimes do not consider the fact the the jcr they are looking at is not initialized completely (and there is not really a good way to check for that; some do it by checking for jobid==0 but thats not great).

One  such example that does nothing to check for initialization is in `ListJobsWaitingOnReservation` in `stored/status.cc`:

```C++
  foreach_jcr (jcr) {
    if (!jcr->sd_impl->reserve_msgs) { continue; } // <- jcr->sd_impl might still be 0 here!
    SendDriveReserveMessages(jcr, sp);
  }
```

Reply:
Its probably best to only add jcrs to the chain that are completely set up (jobid and all) but this might break stuff so for now i just made sure that at least the `*_impl` pointers are setup correctly.